pkg/model/issuerdiscovery.go

```diff
@@ -97,7 +98,15 @@ func (b *IssuerDiscoveryModelBuilder) Build(c *fi.ModelBuilderContext) error {
 			publicFileACL = nil
 		}
 	case *vfs.S3Path:
-		// ok
+		isPublic, err := discoveryStore.IsBucketPublic(ctx)
+		if err != nil {
+			return fmt.Errorf("checking if bucket was public: %w", err)
+		}
+		if !isPublic {
+			klog.Infof("serviceAccountIssuers bucket %q is not public, will use object ACL", discoveryStore.Bucket())
```

Comment:
Can simplify by putting `publicFileACL = fi.PtrTo(true)` here, then define it above with `var publicFileACL *bool`